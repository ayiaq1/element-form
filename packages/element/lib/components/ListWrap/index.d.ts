import React from 'react';
import type { IListWrapProps } from '../../typings';
declare const _default: React.MemoExoticComponent<{
  (props: IListWrapProps): JSX.Element;
  displayName: string;
}>;
export default _default;
