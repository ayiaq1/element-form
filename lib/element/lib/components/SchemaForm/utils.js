/*
 * @Author: yihuang
 * @Date: 2022-03-01 11:47:38
 * @Description: 转换工具
 * @LastModifiedBy: yihuang
 */
define(['require', 'exports'], function (require, exports) {
  'use strict';
  Object.defineProperty(exports, '__esModule', { value: true });
  exports.converChangeEvent = void 0;
  /** 转换所有UI组件的onChange数据 */
  var converChangeEvent = function (params) {
    var str;
    if (params === null || params === void 0 ? void 0 : params.target) {
      str = params.target.value;
    } else {
      str = params;
    }
    return str;
  };
  exports.converChangeEvent = converChangeEvent;
});
//# sourceMappingURL=utils.js.map
