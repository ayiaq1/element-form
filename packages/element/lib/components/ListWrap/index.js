var __assign =
  (this && this.__assign) ||
  function () {
    __assign =
      Object.assign ||
      function (t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
          s = arguments[i];
          for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
        }
        return t;
      };
    return __assign.apply(this, arguments);
  };
define(['require', 'exports', 'react', '../Field'], function (require, exports, react_1, Field_1) {
  'use strict';
  Object.defineProperty(exports, '__esModule', { value: true });
  var ListWrap = function (props) {
    var value = props.value,
      className = props.className,
      style = props.style;
    return react_1.default.createElement(
      react_1.default.Fragment,
      null,
      value instanceof Array && value.length
        ? react_1.default.createElement(
            'ul',
            { className: className, style: style },
            value === null || value === void 0
              ? void 0
              : value.map(function (text, index) {
                  return react_1.default.createElement(
                    'li',
                    { key: index },
                    react_1.default.createElement(Field_1.default, __assign({}, props)),
                  );
                }),
          )
        : null,
    );
  };
  ListWrap.displayName = 'ListWrap';
  exports.default = (0, react_1.memo)(ListWrap);
});
//# sourceMappingURL=index.js.map
