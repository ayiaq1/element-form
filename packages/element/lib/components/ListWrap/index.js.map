{"version":3,"file":"index.js","sourceRoot":"","sources":["../../../src/components/ListWrap/index.tsx"],"names":[],"mappings":";;;;;;;;;;;;;;IAUA,IAAM,QAAQ,GAAG,UAAC,KAAqB;QAC7B,IAAA,KAAK,GAAuB,KAAK,MAA5B,EAAE,SAAS,GAAY,KAAK,UAAjB,EAAE,KAAK,GAAK,KAAK,MAAV,CAAW;QAC1C,OAAO,CACL,8DACG,KAAK,YAAY,KAAK,IAAI,KAAK,CAAC,MAAM,CAAC,CAAC,CAAC,CACxC,sCAAI,SAAS,EAAE,SAAS,EAAE,KAAK,EAAE,KAAY,IACzC,KAAkB,aAAlB,KAAK,uBAAL,KAAK,CAAe,GAAG,CAAC,UAAC,IAAI,EAAE,KAAK;YACpC,OAAO,CACL,sCAAI,GAAG,EAAE,KAAK;gBACZ,8BAAC,eAAK,eAAK,KAAK,EAAI,CACjB,CACN,CAAC;QACJ,CAAC,CAAC,CACC,CACN,CAAC,CAAC,CAAC,IAAI,CACP,CACJ,CAAC;IACJ,CAAC,CAAC;IACF,QAAQ,CAAC,WAAW,GAAG,UAAU,CAAC;IAClC,kBAAe,IAAA,YAAI,EAAC,QAAQ,CAAC,CAAC"}