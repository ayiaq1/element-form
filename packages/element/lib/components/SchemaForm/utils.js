/*
 * @Author: yihuang
 * @Date: 2022-03-01 11:47:38
 * @Description: 转换工具
 * @LastModifiedBy: yihuang
 */
/** 转换所有UI组件的onChange数据 */
export var converChangeEvent = function (params) {
  var str;
  if (params === null || params === void 0 ? void 0 : params.target) {
    str = params.target.value;
  } else {
    str = params;
  }
  return str;
};
