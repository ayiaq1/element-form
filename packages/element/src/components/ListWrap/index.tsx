/*
 * @Author: yihuang
 * @Date: 2021-12-02 14:11:36
 * @Description: 列表包裹器 编辑：input，只读：text
 * @LastModifiedBy: yihuang
 */
import React, { memo } from 'react';
import type { IListWrapProps } from '../../typings';
import Field from '../Field';

const ListWrap = (props: IListWrapProps) => {
  const { value, className, style } = props;
  return (
    <>
      {value instanceof Array && value.length ? (
        <ul className={className} style={style as any}>
          {(value as string[])?.map((text, index) => {
            return (
              <li key={index}>
                <Field {...props} />
              </li>
            );
          })}
        </ul>
      ) : null}
    </>
  );
};
ListWrap.displayName = 'ListWrap';
export default memo(ListWrap);
