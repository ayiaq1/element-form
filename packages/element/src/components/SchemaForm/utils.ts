/*
 * @Author: yihuang
 * @Date: 2022-03-01 11:47:38
 * @Description: 转换工具
 * @LastModifiedBy: yihuang
 */

/** 转换所有UI组件的onChange数据 */
export const converChangeEvent = (params: any): any => {
  let str;
  if (params?.target) {
    str = params.target.value;
  } else {
    str = params;
  }
  return str;
};
