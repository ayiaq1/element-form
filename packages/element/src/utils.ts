import {
  FormatTimeType,
  Unix,
  UnixValueOf,
  MOMENT,
  HMS,
  DataType,
  IConverTimeType,
} from './typings';
import moment, { isMoment } from 'moment';
// 转换为 日期的格式化值
const formatMomentFunc = (item: any, format: FormatTimeType, convertResult?: IConverTimeType) => {
  switch (format) {
    case Unix:
      // 时间戳 秒
      return isMoment(item) ? item.unix() : moment(item).unix();
    case UnixValueOf:
      // 时间戳 毫秒
      return isMoment(item) ? item.valueOf() : moment(item).valueOf();
    case MOMENT:
      // moment
      return isMoment(item) ? item : moment(item);
    case HMS:
      return isMoment(item) ? item.format(HMS) : moment(item, HMS).format(HMS);
    default:
      if (typeof item === 'string') {
        return item;
      } else if (isMoment(item)) {
        return item.format((convertResult as DataType) ?? format);
      } else if (Number.isInteger(item)) {
        return moment(item).unix();
      } else {
        return moment(item).format(format);
      }
    // return typeof item === 'string'
    //   ? item
    //   : isMoment(item)
    //   ? item.format((convertResult as DataType) ?? format)
    //   : Number.isInteger(item)
    //   ? moment.unix(item)
    //   : moment(item).format(format);
  }
};
export { formatMomentFunc };
