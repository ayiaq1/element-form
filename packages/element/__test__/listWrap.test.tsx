/*
 * @Author: yihuang
 * @Date: 2022-03-23 11:13:41
 * @Description: listWrap 组件
 * @LastModifiedBy: yihuang
 */
import React from 'react';
import { mount } from 'enzyme';
import mountTest from '../tests/mountTest';
import renderer from 'react-test-renderer';
import ListWrap from '../src/components/ListWrap';
import { waitForComponentToPaint } from '../tests/utils';

/* Snapshot is a tool to test the code. */
describe('ListWrap', () => {
  mountTest(ListWrap);

  it('Save ListWrap Input Snapshot', async () => {
    const wrapper = renderer.create(<ListWrap mode="edit" value={['text', 'text1']} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('Save ListWrap Text Snapshot', async () => {
    const wrapper = renderer.create(<ListWrap mode="read" value={['text', 'text1']} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('default value', async () => {
    const value = ['default', 'default1', 'default2'];
    const wrapper = mount(<ListWrap mode="edit" value={value} />);
    expect(wrapper.find('li').length).toBe(3);
  });

  it('onChange event', async () => {
    const onChange = jest.fn();
    const value = ['default', 'default1', 'default2'];
    const wrappers = mount(<ListWrap mode="edit" value={value} onChange={onChange} />);
    wrappers.find('li').forEach(async (wrapper, index) => {
      const input = wrapper.find('input');
      const i = index + 1;
      const ChangeValue = `${value}__${i}`;
      input.simulate('change', { target: { value: ChangeValue } });
      await waitForComponentToPaint(wrapper);
      expect(onChange).toHaveBeenCalledTimes(value.length + i);
      expect(onChange).toHaveBeenLastCalledWith(ChangeValue);
    });
  });
});
