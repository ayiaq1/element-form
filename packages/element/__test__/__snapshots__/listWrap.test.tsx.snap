// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListWrap Save ListWrap Input Snapshot 1`] = `
<ul>
  <li>
    <span
      className="ant-input-affix-wrapper"
      onMouseUp={[Function]}
    >
      <input
        className="ant-input"
        hidden={null}
        maxLength={200}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        style={null}
        type="text"
        value="text,text1"
      />
      <span
        className="ant-input-suffix"
      >
        <span
          className="ant-input-clear-icon"
          onClick={[Function]}
          onMouseDown={[Function]}
          role="button"
          tabIndex={-1}
        >
          <span
            aria-label="close-circle"
            className="anticon anticon-close-circle"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="close-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
              />
            </svg>
          </span>
        </span>
      </span>
    </span>
  </li>
  <li>
    <span
      className="ant-input-affix-wrapper"
      onMouseUp={[Function]}
    >
      <input
        className="ant-input"
        hidden={null}
        maxLength={200}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        style={null}
        type="text"
        value="text,text1"
      />
      <span
        className="ant-input-suffix"
      >
        <span
          className="ant-input-clear-icon"
          onClick={[Function]}
          onMouseDown={[Function]}
          role="button"
          tabIndex={-1}
        >
          <span
            aria-label="close-circle"
            className="anticon anticon-close-circle"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="close-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
              />
            </svg>
          </span>
        </span>
      </span>
    </span>
  </li>
</ul>
`;

exports[`ListWrap Save ListWrap Text Snapshot 1`] = `
<ul>
  <li>
    <span
      className="ant-typography ant-typography-ellipsis ant-typography-single-line"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
      text
      text1
      <span
        aria-hidden={true}
        style={
          Object {
            "display": "block",
            "left": 0,
            "pointerEvents": "none",
            "position": "fixed",
            "top": 0,
            "visibility": "hidden",
            "whiteSpace": "nowrap",
            "wordBreak": "keep-all",
            "zIndex": -9999,
          }
        }
      >
        lg
      </span>
      <span
        aria-hidden={true}
        style={
          Object {
            "display": "block",
            "left": 0,
            "margin": 0,
            "padding": 0,
            "pointerEvents": "none",
            "position": "fixed",
            "top": 0,
            "visibility": "hidden",
            "whiteSpace": "normal",
            "width": 0,
            "zIndex": -9999,
          }
        }
      >
        <span
          aria-hidden={true}
        >
          ...
        </span>
      </span>
    </span>
  </li>
  <li>
    <span
      className="ant-typography ant-typography-ellipsis ant-typography-single-line"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
      text
      text1
      <span
        aria-hidden={true}
        style={
          Object {
            "display": "block",
            "left": 0,
            "pointerEvents": "none",
            "position": "fixed",
            "top": 0,
            "visibility": "hidden",
            "whiteSpace": "nowrap",
            "wordBreak": "keep-all",
            "zIndex": -9999,
          }
        }
      >
        lg
      </span>
      <span
        aria-hidden={true}
        style={
          Object {
            "display": "block",
            "left": 0,
            "margin": 0,
            "padding": 0,
            "pointerEvents": "none",
            "position": "fixed",
            "top": 0,
            "visibility": "hidden",
            "whiteSpace": "normal",
            "width": 0,
            "zIndex": -9999,
          }
        }
      >
        <span
          aria-hidden={true}
        >
          ...
        </span>
      </span>
    </span>
  </li>
</ul>
`;
